services/galley/src/Galley/API/Teams.hs

```diff
@@ -141,7 +145,10 @@ uncheckedDeleteTeam zusr zcon tid = do
         pushSome ((newPush1 zusr (TeamEvent e) r & pushConn .~ zcon) : events)
         when ((view teamBinding . Data.tdTeam <$> team) == Just Binding) $
             mapM_ (deleteUser . view userId) membs
-        Data.deleteTeam tid
+        rsp <- Data.deleteTeam tid
+        when ((view teamBinding . Data.tdTeam <$> team) == Just Binding) $
+            journal $ Journal.teamDelete tid
+        return rsp
```

Comment:
Shouldn't we reverse this, i.e., `deleteTeam`should probably be done after the journaling as that'd guarantee that we journal the event at least once